compute_tools/src/compute.rs

```diff
@@ -1367,6 +1368,46 @@ LIMIT 100",
         download_size
     }
 
+    pub async fn set_role_grants(
```

Comment:
It'd be nice to test this flow.

It can be a very simple test:
- start compute
- create role
- add some grants via API that you expect to add in prod
- check with sql that grants were successfully added
- maybe test some edge cases, like try to add the same grants twice
- (optional) on role deletion some grants might prevent the deletion, so we have some code that reassigns owned objects and drops privileges before deletion. I think it should work, but properly it'd be great to also try to delete such role via `compute_ctl`. To do that, you need to restart compute/endpoint with slightly modified spec via `respec()` helper and by adding json
   ```json
    "delta_operations": [
        {
            "action": "delete_role",
            "name": "test_role"
        }
    ]
   ```
   I think that should work

As an example, you could reference to https://github.com/neondatabase/neon/blob/8a114e3aeda7a2e321fa4524335c1748448cae07/test_runner/regress/test_installed_extensions.py#L6 and commit that added it